peer/peer_storage.go

```diff
@@ -0,0 +1,118 @@
+package peer
+
+import (
+	"errors"
+
+	"github.com/lightningnetwork/lnd/kvdb"
+)
+
+var (
+	// peerStorage is the key used in kvdb to store backup data received
+	// from peers.
+	peerStorage = []byte("peer-storage")
+
+	// ErrUninitializedDB signifies an error encountered when attempting
+	// to access the database before it has been set up.
+	ErrUninitializedDB = errors.New("uninitialized Kvdb peer data store")
+)
+
+// PeerStorageProducer creates a backup storage instance for a peer.
+//
+//nolint:revive
+type PeerStorageProducer struct {
+	// DB represents the key-value database backend.
+	DB kvdb.Backend
+}
+
+// NewPeerStorageProducer creates a new PeerStorageProducer with the given
+// database backend.
+func NewPeerStorageProducer(dB kvdb.Backend) (*PeerStorageProducer, error) {
+	return &PeerStorageProducer{
+		DB: dB,
+	}, nil
+}
+
+// InitPeerStorageDB initializes the peer storage database by creating a
+// top-level bucket.
+func InitPeerStorageDB(dB kvdb.Backend) error {
```

Comment:
hmm i think it is strange that this doesnt return an initialised db. See `wdb.OpenTowerDB` -> it does the initialisation and then returns the initialised DB and then uses that to set `dbs.TowerServerDB`. You'd want something similar here

Reply:
I think I should follow same pattern but is there a particular reason why the database was wrapped?

Reply:
Yeah I'm not a fan of "init" methods that don't return the resource being initialized.

Reply:
Please fix this. Take the contents of this method and put it into the body of NewPeerStorageProducer.

Reply:
@ProofOfKeags this is context as to why that was added: https://github.com/lightningnetwork/lnd/pull/8490#discussion_r1528411766